File: primary-algorithm/src/main/java/moon/scala/MoveZeroes.scala
package moon.scala

class MoveZeroes {
  def moveZeroes(nums:Array[Int]):Array[Int]={
    var i=0
    var j=0
    while(j<nums.length&&i<nums.length){
      while(j<nums.length&&nums(j)==0){
        j+=1
      }
      while(i<nums.length&&nums(i)!=0){
        i+=1
      }
      if(i<nums.length&&j<nums.length){
        if(nums(j)!=0&&nums(i)==0&&j>i){
          val a=nums(i)
          nums(i)=nums(j)
          nums(j)=a
        }else j=i
      }
    }
    nums
  }
}

File: data-structure/src/main/java/array/MoveZeroes.java
package array;

public class MoveZeroes {
    public void solve(int[] nums) {
        int node=0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]==0){
                continue;
            }
            nums[node++]=nums[i];
        }
        for(int i=node;i<nums.length;i++){
            nums[i]=0;
        }
    }
}
